### ISSUE TITLE
webpack + mongoose 5.0.1 error and findByIdAndRemove error on 4.13.9

### ISSUE THREAD

**Issue by fazelmk** [Automattic/mongoose#6019]
```
Syntax error in module 'delete': SyntaxError
collection = new !(function webpackMissingModule() { var e = new Error("Cannot find module \".\""); e.code = 'MODULE_NOT_FOUND'; throw e; }())('./db')(
^
SyntaxError: Unexpected token !
at createScript (vm.js:56:10)
at Object.runInThisContext (vm.js:97:10)
at Module._compile (module.js:542:28)
at Object.Module._extensions..js (module.js:579:10)
at Module.load (module.js:487:32)
at tryModuleLoad (module.js:446:12)
at Function.Module._load (module.js:438:3)
at Module.require (module.js:497:17)
at require (internal/module.js:20:19)
```

webpack:  3.10.0

I tried to update mongoose to see if a bug on findByIdAndRemove was already fixed. And got that error.

The problem in findByIdAndRemove is that it sometimes ( ~60% of the time ) return error = null, document = null on the callback. But It finds the document with the Id and remove it. If I look the collection the document is removed but it returns a null document. What makes my function return a 404 error. That is not true because the document exist, was found and removed successfully. 

my package.json working but with problem in findByIdAndRemove

```
{
  "name": "aws-nodejs-typescript",
  "version": "1.0.0",
  "description": "Serverless webpack example using Typescript",
  "main": "handler.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "devDependencies": {
    "@types/aws-lambda": "0.0.22",
    "@types/mongodb": "^2.2.16",
    "@types/mongoose": "^4.7.28",
    "@types/node": "^8.0.55",
    "aws-lambda": "^0.1.2",
    "aws-lambda-typescript": "^1.0.1",
    "aws-sdk": "^2.163.0",
    "serverless": "git+https://github.com/hgiasac/serverless.git#sls-agw-authorizer",
    "serverless-offline": "^3.16.0",
    "serverless-plugin-swag": "git+https://github.com/doapp-ryanp/serverless-plugin-swag.git",
    "serverless-prune-plugin": "^1.3.0",
    "serverless-webpack": "^3.1.2",
    "ts-loader": "^2.3.7",
    "tslint": "^5.8.0",
    "tslint-config-olo": "^0.2.0",
    "typescript": "^2.6.2",
    "webpack": "^3.10.0"
  },
  "author": "Fazel Mobini Kesheh",
  "license": "",
  "dependencies": {
    "mongo-sanitize": "^1.0.0",
    "mongoose": "^4.13.9",
    "mongoose-id-validator": "^0.4.3"
  }
}
```

my function

```
let Model = this.mongoose.model( "MyModel", new Schema({name: string}) );
Model.findByIdAndRemove( this.mongoose.Types.ObjectId( id ), ( err: any, document: any ) => {
  if ( err ) {
    return reject( { statusCode: 500, body: { error: err } } );
  }
  if ( !document ) {
    return reject( { statusCode: 404, body: { error: "Resource not found" } } );
  }

  return resolve( { statusCode: 200, body: { message: "Resource successfully removed!", data: document } } );
} ).catch( ( err: any ) => {
  console.error( err );
  return reject( { statusCode: 500, body: { error: err } } );
} );
```

**Comment by vkarpov15** [comment]
Is it possible that your function is getting executed multiple times? Like hitting the API endpoint multiple times?

Also, what does that syntax error have to do with this issue?

**Comment by MuhammadShaban** [comment]
this problem appears after update to mongoose 5.x
the problem with mongodb package in this file mongodb/lib/collection.js
`
   2997       collection = new require('./db')(
   2998         doc.db,
   2999         self.s.db.s.topology,
   3000         self.s.db.s.options
   3001       ).collection(doc.collection);
`
see this issues [https://github.com/webpack/webpack/issues/6191](url) for more information

**Comment by fazelmk** [comment]
@vkarpov15 checked cloudWatch and the endpoint was hit only once.

**Comment by fazelmk** [comment]
will wait on mongodb update to update to 5.x

**Comment by varunjayaraman** [comment]
@vkarpov15 it looks like all we need to do is upgrade to node-mongodb-native 3.0.2 once that patch is released.

**Comment by varunjayaraman** [comment]
@fazelmk are you saying your findByIdAndRemove is removing the document but `document` is null in the callback nonetheless?

**Comment by fazelmk** [comment]
@varunjayaraman Yes. that is what happens, sometimes it just returns error as null and document as null even though it did remove the document I requested to be removed. I did try model.remove( id ) but sometimes it returned n as 0, saying it didn't change any record but it too where removing the recorded. 

Will update to latest version as soon as you mongodb-native guys deploy the fix and you guys update the dependency.

If I catch the problem again in the new version then we can look at it. No point in wasting time in a problema that could have been already fixed.
